### PR TITLE
[script.logviewer] 2.1.1

### PR DESCRIPTION
**Pull request by i96751414**
### Add-on details:

- **General**
  - Add-on name: Log Viewer for Kodi
  - Add-on ID: script.logviewer
  - Version number: 2.1.1
  - Kodi/repository version: gotham

- **Code location**
  - URL: 
  
Tools to easily check and analyse Kodi log file.

### Description of changes:

- Python 3 compatibility
- Fix log error detection service after kodi's migration to libfmt
- Fix android file handling
- Add http server (webtail)
        

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: script.logviewer/resources/language/Portuguese/strings.po [hunk 1]
```diff
@@ -112,3 +112,13 @@ msgctxt "#30018"
...
 ""
+
+msgctxt "#30019"
+msgid "Enable HTTP server"
+msgstr "Activar servidor HTTP"
+""
+
+msgctxt "#30020"
+msgid "Port"
+msgstr "Porta"
```
Comment (enen92) [inline script.logviewer/resources/language/Portuguese/strings.po]: Porto :)

File: script.logviewer/resources/language/Portuguese (Brazil)/strings.po [hunk 2]
```diff
@@ -112,3 +112,13 @@ msgctxt "#30018"
...
 ""
+
+msgctxt "#30019"
+msgid "Enable HTTP server"
+msgstr "Activar servidor HTTP"
+""
+
+msgctxt "#30020"
+msgid "Port"
+msgstr "Porta"
```
Comment (enen92) [inline script.logviewer/resources/language/Portuguese (Brazil)/strings.po]: Porto

File: script.logviewer/resources/lib/httpserver.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import os
+import shutil
+import sys
+import threading
+
+from resources.lib.logreader import LogReader
+from resources.lib.logviewer import log_location
+from resources.lib.utils import ADDON_PATH, encode
+
+if sys.version_info.major >= 3:
```
Comment (enen92) [inline script.logviewer/resources/lib/httpserver.py]: IIRC you can just import `PY3` from utils

File: script.logviewer/resources/lib/navigation.py [hunk 4]
```diff
@@ -12,6 +12,11 @@ def has_addon(addon_id):
     return xbmc.getCondVisibility("System.HasAddon({})".format(addon_id)) == 1
 
 
+def test_exception():
```
Comment (enen92) [inline script.logviewer/resources/lib/navigation.py]: leftover?

File: script.logviewer/resources/lib/service.py [hunk 5]
```diff
@@ -1,67 +1,117 @@
...
-    def restart(self):
-        self.stop()
-        self.start()
+        self._running = False
+        self._stop_error_popup_runner()
+        self._stop_http_server_runner()
+
+    def _start_error_popup_runner(self):
+        if self._error_popup_runner is None:
+            logging.info("Starting error popup runner")
```
Comment (enen92) [inline script.logviewer/resources/lib/service.py]: change to debug please, default log level

File: script.logviewer/resources/lib/service.py [hunk 6]
```diff
@@ -1,67 +1,117 @@
...
+            self._error_popup_runner = ErrorPopupRunner(self)
+            self._error_popup_runner.start()
+
+    def _stop_error_popup_runner(self):
+        if self._error_popup_runner is not None:
+            logging.info("Stopping error popup runner")
```
Comment (enen92) [inline script.logviewer/resources/lib/service.py]: Same

File: script.logviewer/resources/lib/service.py [hunk 7]
```diff
@@ -1,67 +1,117 @@
...
+        self._monitor = monitor
+        self._port = port
+        self._server = None
+        super(HTTPServerRunner, self).__init__()
+
+    def run(self):
+        self._server = ThreadedHTTPServer(("", self._port), ServerHandler)
+        self._server.daemon_threads = True
+
+        logging.info("Server started at port {}".format(self._port))
```
Comment (enen92) [inline script.logviewer/resources/lib/service.py]: debug

File: script.logviewer/resources/lib/utils.py [hunk 8]
```diff
@@ -1,39 +1,58 @@
...
+    def decode(s):
+        return s
+
 
 def get_setting(setting):
     return ADDON.getSetting(setting)
 
 
-def get_boolean(setting):
+def get_boolean_setting(setting):
```
Comment (enen92) [inline script.logviewer/resources/lib/utils.py]: Kodi provides methods to get settings as bools in newest releases (Leia): `getSettingBool`, `getSettingInt`, etc, see: https://codedocs.xyz/xbmc/xbmc/group__python__xbmcaddon.html#gaf0205dd5b96354f30a39fbf35599e681

Maybe you can do some sort of runtime version detection and use those instead

File: script.logviewer/addon.xml [hunk 9]
```diff
@@ -21,6 +21,8 @@
         <news>
 - Python 3 compatibility
 - Fix log error detection service after kodi's migration to libfmt
+- Fix android file handling
```
Comment (enen92) [inline script.logviewer/addon.xml]: News should be only for the newest release. You're including the changes from the previous version too

**Comment by enen92** [comment]
Btw, @i96751414 to update the pull request you just need to remove the tag, commit your changes and tag it again without closing this pull request. `kodi-addon-submitter` will detect the already opened PR and update it instead of creating a new one.

File: script.logviewer/resources/lib/navigation.py (hunk 4 above)
Comment (i96751414) [reply script.logviewer/resources/lib/navigation.py]: It was left on purpose, so I don't have to write it every time I want to manually test an exception.

File: script.logviewer/resources/lib/utils.py (hunk 8 above)
Comment (i96751414) [reply script.logviewer/resources/lib/utils.py]: I would prefer to keep these ones so I don't have to do compatibility checks.

**Pull request merged**